File: iris-ld/iris-ld-proofs/src/main/java/city/smartb/iris/ldproof/util/SHAUtil.java
package city.smartb.iris.ldproof.util;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class SHAUtil {

	private SHAUtil() {
	}

	public static byte[] sha256(String string) {
		try {
			MessageDigest digest = MessageDigest.getInstance("SHA-256");
			return digest.digest(string.getBytes(StandardCharsets.UTF_8));
		} catch (NoSuchAlgorithmException ex) {
			throw new RuntimeException(ex.getMessage(), ex);
		}


	}
}

File: iris-ld/iris-ld-proofs/src/main/kotlin/city/smartb/iris/ld/ldproof/util/SHAUtil.kt
package city.smartb.iris.ld.ldproof.util

import java.nio.charset.StandardCharsets
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

object SHAUtil {
    fun sha256(string: String?): ByteArray {
        return try {
            val digest = MessageDigest.getInstance("SHA-256")
            digest.digest(string!!.toByteArray(StandardCharsets.UTF_8))
        } catch (ex: NoSuchAlgorithmException) {
            throw RuntimeException(ex.message, ex)
        }
    }
}
